File: Leetcode 2265. Count Nodes Equal to Average of Subtree/lc 2265.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    int ans=0;
    public int averageOfSubtree(TreeNode root) {
        traverse(root);
        return ans;
    }

    int[] traverse(TreeNode node){
        if(node==null) {
            int[] arr={0,0};
            return arr;
        }
        int[] left=traverse(node.left);
        int[] right=traverse(node.right);
        int sum=node.val+left[0]+right[0];
        int count=1+left[1]+right[1];
        int avg=sum/count;
        if(avg==node.val)ans++;
        int[] arr={sum,count};
        return arr;

    }
}

File: Leetcode 2265. Count Nodes Equal to Average of Subtree/lc 2265.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     T,reeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int ans=0;
    pair<int,int> traverse(TreeNode *node){
        if(node==nullptr)return {0,0};
        pair<int,int> left=traverse(node->left);
        pair<int,int> right=traverse(node->right);
        int sum=node->val+left.first+right.first;
        int count=1+left.second+right.second;
        int avg=sum/count;
        if(avg==node->val)ans++;
        return {sum,count};
    }
    int averageOfSubtree(TreeNode* root) {
        traverse(root);
        return ans;
    }
};